knowledge-base/window-keep-content-when-closed.md

```diff
@@ -0,0 +1,222 @@
+
+3. In the `OnClick` handler of the custom close action, set a custom CSS class to the Window through the `Class` parmeter.
+
+4. Use this class to add `display: none;` rule to the Window.
+
+5. Clear this class when the user opens the Window to revert the custom style and display the componet.
+
+````CSHTML
+@*Hide Window with CSS to preserve its content*@
+
+@using System.ComponentModel.DataAnnotations
+
+<style>
+    .hidden-window {
+        display: none;
+    }
+</style>
+
+<TelerikWindow @bind-Visible="@WindowRendered"
+               Class="@WindowClass">
+    <WindowTitle>
+        Window Title
+    </WindowTitle>
+    <WindowContent>
+
+        <TelerikForm Model="@person">
+            <FormValidation>
+                <DataAnnotationsValidator />
+            </FormValidation>
+        </TelerikForm>
+
+    </WindowContent>
+    <WindowActions>
+        <WindowAction Name="CustomClose"
+                      Icon="@FontIcon.X"
+                      OnClick="@CustomCloseHandler" />
+    </WindowActions>
+</TelerikWindow>
+
+<TelerikButton OnClick="@OpenWindowHandler">Open Window</TelerikButton>
+
+@code {
+    private bool WindowRendered { get; set; }
+
+    private string WindowClass { get; set; }
+
+    private void CustomCloseHandler()
+    {
+        WindowClass = "hidden-window";
+    }
+
+    private void OpenWindowHandler()
+    {
+        if (!WindowRendered)
+        {
+            WindowRendered = true;
+        }
+
+        WindowClass = "";
+    }
+
+    private Person person = new Person();
+
+    public class Person
+    {
+        [Required]
+        public int Id { get; set; }
+
+        [Required]
+        [Display(Name = "First Name")]
+        public string FirstName { get; set; }
+
+        [Required]
+        [Display(Name = "Last Name")]
+        public string LastName { get; set; }
+
+        [Required]
+        [Display(Name = "Date of Birth")]
+        public DateTime? DOB { get; set; }
+    }
+}
+````
+
+### Custom close for modal Window
+
+The [modal Window]({%slug components/window/modal%}) has a different rendering. It is wrapped in a `<div class="k-dialog-wrapper">` which contains both the Window and the overlay. To allow the user interact with the page while the Winwod is hidden you also need to hide the overlay.
+
+The next steps for custom closing a modal Window are:
+
+3. In the `OnClick` handler of the custom close action, toggle a flag to indicate the user wants to hide the Window.
+
+4. Based on this flag value, conditionally apply `display: none;` rule to the `<div class="k-dialog-wrapper">`. This will hide both the Window and the overlay.
+
+5. Toggle back the flag when the user opens the Window to revert the custom style and display the component.
+
+````CSHTML
+@*Hide modal Window with CSS to preserve its content*@
+
+@using System.ComponentModel.DataAnnotations
+
+@if (WindowHidden)
+{
+    <style>
+        .k-dialog-wrapper {
+            display: none;
```

Comment:
You can mention that while this approach is "active", the application cannot use other modal Windows or Dialogs.